codec-classes-quic/src/main/java/io/netty/handler/codec/quic/QuicChannel.java

```diff
@@ -55,97 +53,97 @@
-    default ChannelFuture connect(SocketAddress remoteAddress, SocketAddress localAddress) {
+    default Future<Void> connect(SocketAddress remoteAddress, SocketAddress localAddress) {
         return pipeline().connect(remoteAddress, localAddress);
     }
 
     @Override
-    default ChannelFuture disconnect() {
+    default Future<Void> disconnect() {
         return pipeline().disconnect();
     }
 
     @Override
-    default ChannelFuture close() {
+    default Future<Void> close() {
         return pipeline().close();
     }
 
     @Override
-    default ChannelFuture deregister() {
+    default Future<Void> deregister() {
         return pipeline().deregister();
     }
 
     @Override
-    default ChannelFuture bind(SocketAddress localAddress, ChannelPromise promise) {
+    default Future<Void> bind(SocketAddress localAddress, Promise<Void> promise) {
         return pipeline().bind(localAddress, promise);
     }
 
     @Override
-    default ChannelFuture connect(SocketAddress remoteAddress, ChannelPromise promise) {
+    default Future<Void> connect(SocketAddress remoteAddress, Promise<Void> promise) {
         return pipeline().connect(remoteAddress, promise);
     }
 
     @Override
-    default ChannelFuture connect(SocketAddress remoteAddress, SocketAddress localAddress, ChannelPromise promise) {
+    default Future<Void> connect(SocketAddress remoteAddress, SocketAddress localAddress, Promise<Void> promise) {
         return pipeline().connect(remoteAddress, localAddress, promise);
     }
 
     @Override
-    default ChannelFuture disconnect(ChannelPromise promise) {
+    default Future<Void> disconnect(Promise<Void> promise) {
         return pipeline().disconnect(promise);
     }
 
     @Override
-    default ChannelFuture close(ChannelPromise promise) {
+    default Future<Void> close(Promise<Void> promise) {
         return pipeline().close(promise);
     }
 
     @Override
-    default ChannelFuture deregister(ChannelPromise promise) {
+    default Future<Void> deregister(Promise<Void> promise) {
         return pipeline().deregister(promise);
     }
 
     @Override
-    default ChannelFuture write(Object msg) {
+    default Future<Void> write(Object msg) {
         return pipeline().write(msg);
     }
 
     @Override
-    default ChannelFuture write(Object msg, ChannelPromise promise) {
+    default Future<Void> write(Object msg, Promise<Void> promise) {
         return pipeline().write(msg, promise);
     }
 
     @Override
-    default ChannelFuture writeAndFlush(Object msg, ChannelPromise promise) {
+    default Future<Void> writeAndFlush(Object msg, Promise<Void> promise) {
         return pipeline().writeAndFlush(msg, promise);
     }
 
     @Override
-    default ChannelFuture writeAndFlush(Object msg) {
+    default Future<Void> writeAndFlush(Object msg) {
         return pipeline().writeAndFlush(msg);
     }
 
     @Override
-    default ChannelPromise newPromise() {
+    default Promise<Void> newPromise() {
```

Comment:
This could be return type generic, like:

```java
default <V> Promise<V> newPromise() {
```

Similar for succeeded and failed method variants.